graph/pull.go

```diff
@@ -352,9 +352,9 @@ func (s *TagStore) pullRepository(r *registry.Session, out io.Writer, repoInfo *
 		}
 	}
 
-	requestedTag := repoInfo.CanonicalName
+	requestedTag := repoInfo.LocalName
 	if len(askedTag) > 0 {
-		requestedTag = utils.ImageReference(repoInfo.CanonicalName, askedTag)
```

Comment:
@dmcgowan it's not exactly obvious to me why this was CanonicalName and why it's being changed now. Other than that, the PR is cool!